models.go

```diff
@@ -1,60 +1,25 @@
 package main
 
 import (
-	"database/sql"
-	"time"
-
 	"gorm.io/gorm"
 )
 
-// User has one `Account` (has one), many `Pets` (has many) and `Toys` (has many - polymorphic)
-// He works in a Company (belongs to), he has a Manager (belongs to - single-table), and also managed a Team (has many - single-table)
-// He speaks many languages (many to many) and has many friends (many to many - single-table)
-// His pet also has one Toy (has one - polymorphic)
 type User struct {
 	gorm.Model
-	Name      string
-	Age       uint
-	Birthday  *time.Time
-	Account   Account
-	Pets      []*Pet
-	Toys      []Toy `gorm:"polymorphic:Owner"`
-	CompanyID *int
-	Company   Company
-	ManagerID *uint
-	Manager   *User
-	Team      []User     `gorm:"foreignkey:ManagerID"`
-	Languages []Language `gorm:"many2many:UserSpeak"`
-	Friends   []*User    `gorm:"many2many:user_friends"`
-	Active    bool
-}
-
-type Account struct {
-	gorm.Model
-	UserID sql.NullInt64
-	Number string
+	Pets []*Pet
 }
 
 type Pet struct {
 	gorm.Model
-	UserID *uint
 	Name   string
-	Toy    Toy `gorm:"polymorphic:Owner;"`
+	FavToy *Toy
```

Comment:
You probably want
```go
type User struct {
	gorm.Model
	Pets []*Pet
}

type Pet struct {
	gorm.Model
	Name   string
	FavToy *Toy   `gorm:"foreignKey:FavoritedBy;references:ID"`
	Toys   []*Toy `gorm:"foreignKey:ID"`
	UserID uint
}

type Toy struct {
	gorm.Model
	Name        string
	PetID       uint
	FavoritedBy uint
}
```

Reply:
I can see what you're saying with the having the `FavoritedBy` field. The issue with this I feel is that the DB semantics wouldn't be quite right.

For example, it would then be possible within the DB for many of the Pet's toys to have been favourited by the pet. In the modelling I'm going for, I'd like to enforce that a Pet can have only a single favourite however.

I didn't spot that the pets table was missing the fav_toy_id column. I've updated the PR, adding `FavToyID` to the `Pet` struct which fixes that.